server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -226,7 +226,7 @@ private void initialization() throws Exception {
     if (healthCheckEnable) {
       List<Checker> builtInCheckers = Lists.newArrayList();
       builtInCheckers.add(storageManager.getStorageChecker());
-      healthCheck = new HealthCheck(isHealthy, shuffleServerConf, builtInCheckers);
+      healthCheck = new HealthCheck(this, shuffleServerConf, builtInCheckers);
```

Comment:
You can pass ServerStatus as parameter. We can use AtomicReference as the type of ServerStatus.

Reply:
Let me revise it.